tests/load/pipeline/test_replace_disposition.py

```diff
@@ -114,19 +103,41 @@ def items_with_subitems():
                 "name": "sub item 102"
             }]
         }
+        yield dlt.mark.with_table_name(data, "items")
+        yield dlt.mark.with_table_name(data, "other_items")
 
-    @dlt.resource(table_name="items", write_disposition="replace", primary_key="id")
+    @dlt.resource(name="main_resource", write_disposition="replace", primary_key="id")
     def items_without_subitems():
-        yield {
+        data = [{
             "id": 1,
             "name": "item",
             "sub_items": []
-        }
+        }]
+        yield dlt.mark.with_table_name(data, "items")
+        yield dlt.mark.with_table_name(data, "other_items")
 
-    @dlt.resource(table_name="items", write_disposition="replace", primary_key="id")
+    @dlt.resource(name="main_resource", write_disposition="replace", primary_key="id")
     def items_with_subitems_yield_none():
         yield None
         yield None
+        data = [{
+            "id": 1,
+            "name": "item",
+            "sub_items": [{
+                "id": 101,
+                "name": "sub item 101"
+            },{
+                "id": 101,
+                "name": "sub item 102"
+            }]
+        }]
+        yield dlt.mark.with_table_name(data, "items")
+        yield dlt.mark.with_table_name(data, "other_items")
+        yield None
```

Comment:
what a little bother me is that when we added the round robin strategy to pipe iterator, we also forced the iterator to get next value if None is yielded. to None should never get out of the pipe:
```
item = None
            while item is None:
                item = next(gen)
```
so how it is possible that you've got this INSERT VALUE with NULL that we were looking at?
```

Reply:
This happens if yield None is the only statement in the resource and no data is ever yielded. I did not investigate why this happens though.

Reply:
Should I? I think this would be a problem for a separate ticket though and we should merge the replace soonish